python_benchmarks/core.py

```diff
@@ -0,0 +1,202 @@
+import ctypes
+import torch
+from torch.profiler import profile, ProfilerActivity
+from typing import List, Callable, Union
+from torch.autograd import DeviceType
+
+
+def get_device_properties() -> tuple:
+    """
+    Computes L2 cache size and peak device bandwidth using ctypes and cuda.
+    """
+    libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        raise OSError("could not load any of: " + " ".join(libnames))
+
+    # Device attribute enums (taken from cuda.h)
+    # https://nvidia.github.io/cuda-python/module/cuda.html
```

Comment:
I was avoiding the dependency on cuda-python. These constants are from `cuda.h` and haven't changed with driver updates so I am okay with using the enums to avoid the dependency.

Reply:
> I was avoiding the dependency on cuda-python. 

I understand you don't want the initial PR to do too much. But is there a good reason to avoid cuda-python? Is it known to be problematic?

Reply:
There's no strong reason to avoid it, but it's relatively new and no one in our team has any experience with it yet. If it's just for one constant value, I don't feel it would justify adding a new dependency. Also, it says it requires CUDA 12.0 or newer, whereas we still need to support 11.8.

Reply:
I think in the future if we are adding more like this, we should definitely reconsider cuda-python, but probably not right now.

Reply:
> it says it requires CUDA 12.0 or newer, whereas we still need to support 11.8.

That's a good reason. @Priya2698 , for posterity, could you add a comment here saying consider to use cuda-python in the future?

Reply:
Yeah, mostly I was avoiding since we only needed it for these constants and were not using it for any other functionality.